### PR TITLE
[BUG]: index2token change after serialization

### PR DESCRIPTION
**Pull request by SoluMilken**

### CODE REVIEW

File: strpipe/ops/tests/test_pad.py [hunk 1]
```diff
@@ -45,7 +47,8 @@ def test_pad_passing_state_needs_both_sos_and_eos_or_neither():
     ]
 
     state1 = {'maxlen': 0, 'pad_token': '<PAD>'}
-    state2 = {'maxlen': 2, 'pad_token': '<PAD>', 'sos_token': '<sos>', 'eos_token': '<eos>'}
+    state2 = {'maxlen': 2, 'pad_token': '<PAD>',
+              'sos_token': '<sos>', 'eos_token': '<eos>'}
```
Comment (stegben) [inline strpipe/ops/tests/test_pad.py]: Strange coding style. Either one value per line or all in single line

File: strpipe/ops/tests/test_pad.py [hunk 2]
```diff
@@ -84,7 +87,8 @@ def test_pad_passing_state_tokens_in_state_should_be_distinct():
         with pytest.raises(ValueError):
             padder.transform(bad_state, input_data)
 
-    state = {'maxlen': 5, 'pad_token': '<PAD>', 'sos_token': '<sos>', 'eos_token': '<eos>'}
+    state = {'maxlen': 5, 'pad_token': '<PAD>',
+             'sos_token': '<sos>', 'eos_token': '<eos>'}
```
Comment (stegben) [inline strpipe/ops/tests/test_pad.py]: Same.

File: strpipe/ops/tests/test_token_to_index.py [hunk 3]
```diff
@@ -95,12 +101,12 @@ def test_inverse_transform():
...
         state={
             'index2token': {
-                0: unk_token,
-                1: "喜歡",
-                2: "吃",
-                3: "榴槤",
+                '0': unk_token,
+                '1': "喜歡",
+                '2': "吃",
+                '3': "榴槤",
```
Comment (stegben) [inline strpipe/ops/tests/test_token_to_index.py]: Why? The index should be integer.

**Comment by stegben** [comment]
Could you refine the purpose of this bug? What happened when serializing?

**Comment by SoluMilken** [comment]
```python
import json
A = {0: '0', 1: '1'}
with open('789.json', 'w') as filep: 
    json.dump(A, filep, ensure_ascii=False)
def filep
with open('789.json', 'r') as filep: 
    B = json.load(filep)
print(B)  # returns {'0': '0', '1': '1'}
A == B  # returns False
```

File: strpipe/ops/tests/test_pad.py (hunk 1 above)
Comment (SoluMilken) [reply strpipe/ops/tests/test_pad.py]: 4c85da4

File: strpipe/ops/tests/test_pad.py (hunk 2 above)
Comment (SoluMilken) [reply strpipe/ops/tests/test_pad.py]: 4c85da4

File: strpipe/ops/tests/test_pad.py (hunk 1 above)
Comment (SoluMilken) [reply strpipe/ops/tests/test_pad.py]: f210f41

File: strpipe/ops/tests/test_pad.py (hunk 2 above)
Comment (SoluMilken) [reply strpipe/ops/tests/test_pad.py]: f210f41

**Comment by stegben** [comment]
please squash the last two commits

File: strpipe/ops/tests/test_pad.py (hunk 1 above)
Comment (SoluMilken) [reply strpipe/ops/tests/test_pad.py]: ea355b3

File: strpipe/ops/tests/test_pad.py (hunk 2 above)
Comment (SoluMilken) [reply strpipe/ops/tests/test_pad.py]: ea355b3

**Pull request merged**